accel/tcg/translator.c

```diff
@@ -127,11 +104,14 @@ void translator_loop(const TranslatorOps *ops, DisasContextBase *db,
         }
 #endif
         ops->insn_start(db, cpu);
-#if defined(CONFIG_MIPS_LOG_INSTR)
-        if (unlikely(should_log_instr)) {
-            TCGv tpc = tcg_const_tl(db->pc_next);
-            gen_helper_log_instruction(cpu_env, tpc);
-            tcg_temp_free(tpc);
+#ifdef CONFIG_TCG_LOG_INSTR
+        /*
+         * TODO(am2419): could move the commit below tr_insn or would we lose something?
+         */
+        if (qemu_log_instr_enabled(cpu->env_ptr)) {
```

Comment:
Can this use `db->log_instr_enabled` instead of calling the check function again? I believe anything that changes instruction logging state should break out of the translation block.
In that case I'd also be tempted to keep a local `const bool` variable since the compiler probably doesn't know that db->log_instr_enabled is constant.

Reply:
It should definitely use the `db->log_instr_enabled`. I will also use a local `const bool`, nice catch.
I should also cleanup some of the TODOs and commented out stuff.